templates/raspbian_cattlepi/resources/usr/share/initramfs-tools/scripts/cattlepi-base/helpers

```diff
@@ -414,3 +447,24 @@ cattlepi_bring_up_wlan() {
         return 1
 	fi
 }
+
+cattlepi_sdmount() {
+    # Explicitly mount the sdcard partition to /boot
+    # We don't give the caller any other option as of right
+    # now since this partition doesn't need to be mounted
+    # anywhere else and doing so could be dangerous
+    echo "Mounting sdcard to /boot"
+    mkdir /boot
+    umount /boot
+    mount -n -t vfat /dev/mmcblk0p1 /boot
+}
+
+cattlepi_sdmount_ro() {
+    # Explicitly mount the sdcard patition RO to /boot
+    # As above, don't give the caller any other option
+    # since we don't need to and it could be dangerous yadda yadda
+    echo "Mounting sdcard RO to /boot"
+    mkdir /boot
+    umount /boot
```

Comment:
insert a **sync** after this umount

Reply:
again, what I think happens is that you mount the SD card, you do a bunch of downloads / writes to it, but when you switch it to readonly the kernel does not finish writing everything to card (vfat behavior). You end up with partial/corrupted files that fail the md5 check. 

I saw something similar when I was playing with the sdlayout. see here: https://github.com/cattlepi/cattlepi/blob/34a2f6696eb9e6b40a9acb34e5e9d4a7cc840d1e/templates/raspbian_cattlepi/resources/usr/share/initramfs-tools/scripts/cattlepi-base/helpers#L245

you also probably want to sync whever you are using the sysreq-trigger, as for example here: https://github.com/cattlepi/cattlepi/blob/d70a7f512bd4c8ea8aa3bdf68d7fce2741f4be96/templates/raspbian_cattlepi/resources/bin/autoupdate.sh#L19